RFC-0039-cuda-support.md

```diff
@@ -0,0 +1,197 @@
+
+# [CUDA version support]
+
+**Authors:**
+* @atalman @malfet @tinglvv @nWEIdia @ptrblck
+
+
+## **Motivation**
+
+The proposal is to provide two main benefits
+
+- Aligned decision policy provides transparent decision making;
+- Incorporate decision points in the early release process to stagger CUDA migration from the RC release integration work.
+
+## **Proposed Policy and Process**
+
+### We should introduce new version of CUDA when
+
+- It enables new important GPU architecture (For example Blackwell with CUDA-12.8)
+- It brings significant performance improvements
+- It fixes significant correctness issues
+- It significantly reduces binary/memory footprint
+
+### We would deprecate version of CUDA when
+
+As soon as we introduce a new Experimental Version we should consider moving the previous Experimental Version to Stable, and decommission the previous Stable version. Typically we want to support 2-3 versions of CUDA as follows:
+
+- Optional Legacy Version: If we need to have 1 version for backend compatibility or to work around the current limitation. For example: CUDA older driver is incompatible with newer CUDA version
+- Stable Version: This is a stable CUDA version that is used most of the time. This is the version we want to upload to PyPI. Please note that if the stable version is equal or slightly older to the version then the version fbcode is using, we probably need to keep it in CI/CD system as Optional Legacy Version.
```

Comment:
I don't see why we should talk about fbcode here?
If anything we can have partners ask for a temporary legacy version as above